#!/usr/bin/exec-suid --real -- /usr/bin/python3 -I
import sys
sys.path.append('/challenge')

def print_flag():
    try:
        with open("/flag", "r") as f:
            print(f.read())
    except FileNotFoundError:
        print("Error: Flag file not found.")

# Add your imports and other code below here
import os
import re
import subprocess

def main(filename):
    evenNums = "0 2 4 6 8 10 12 14 16 18 20 22 24 26 28 30 ?"
    oddNums = "41 43 45 47 49 51 53 55 57 59 61 63 65 67 69 ?"
    thirdNums = "22 25 28 31 34 37 40 43 46 49 52 55 58 61 64 67 70 73 76 79 82 85 88 91 94 97 100 103 106 109 112 115 118 121 124 127 130 ?"
    number = "47"
    randRange = "".join(str(i) + " " for i in range(0,int(number)+1))

    results = subprocess.run([filename, number], capture_output=True, text=True)

    resultList = results.stdout.rstrip("\n").split("\n")
    successCount = 0

    if re.match(evenNums, resultList[0]):
        successCount += 1
    else:
        print("ERROR: There's a problem with your even numbers.")
    
    if re.match(oddNums, resultList[1]):
        successCount += 1
    else:
        print("ERROR: There's a problem with your odd numbers.")
    
    if re.match(thirdNums, resultList[2]):
        successCount += 1
    else:
        print("ERROR: There's a problem with your 3rd numbers.")
    
    if re.match(randRange, resultList[3]):
        successCount += 1
    else:
        print("ERROR: There's a problem with your random range.")
    
    if int(resultList[4]) >= 0 and int(resultList[4]) <= 15:
        successCount += 1
    else:
        print("ERROR: There's a problem with your random loop.")
    
    if 5 != successCount:
        print("Try again! Make sure you're numbers are formatted correctly!")
    else:
        print("Congratulations, you've passed!")
        print_flag()


if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("ERROR: Missing file to verify, exiting.")
        sys.exit(0)

    if os.path.exists(sys.argv[1]):
        main(sys.argv[1])
    else:
        print(f"ERROR: Cannot find file: {sys.argv[1]}, ensure it exists and try again.")